filecoin-proofs/src/bin/phase2.rs

```diff
@@ -355,492 +436,664 @@ fn create_initial_params<Tree: 'static + MerkleTreeTrait>(
-            path_after,
-            hex::encode(&provided_contribution_hash[..])
-        );
+    info!("converting");
 
-        // If we are verifying the first contribution read both `path_before` and `path_after`
-        // files. For every subsequent verification, move the previous loop's "after" params to this
-        // loop's "before" params then read this loop's "after" params file. This will minimize the
-        // number of expensive parameter file reads.
-        let params_before = match next_params_before.take() {
-            Some(params_before) => params_before,
-            None => {
-                info!("reading 'before' params from disk: {}", path_before);
-                let file = File::open(path_before).unwrap();
-                let mut reader = BufReader::with_capacity(1024 * 1024, file);
-                let start = Instant::now();
-                let params_before = phase2::MPCParameters::read(&mut reader, true).unwrap();
-                info!(
-                    "successfully read 'before' params from disk, dt_read={}s",
-                    start.elapsed().as_secs()
-                );
-                params_before
-            }
-        };
+    info!(
+        "making streamer from small 'before' params: {}",
+        path_before
+    );
 
-        let params_after = {
-            info!("reading 'after' params from disk: {}", path_after);
-            let file = File::open(path_after).unwrap();
-            let mut reader = BufReader::with_capacity(1024 * 1024, file);
-            let start = Instant::now();
-            let params_after = phase2::MPCParameters::read(&mut reader, true).unwrap();
-            info!(
-                "successfully read 'after' params from disk, dt_read={}s",
-                start.elapsed().as_secs()
+    let mut streamer = if param_size.is_large() {
+        panic!("cannot convert large param format");
+    } else {
+        Streamer::new(path_before, read_raw, write_raw).unwrap_or_else(|e| {
+            panic!(
+                "failed to make streamer from small `{}`: {}",
+                path_before, e
             );
-            params_after
-        };
-
-        info!("verifying param transition");
-        let start_verification = Instant::now();
-
-        let calculated_contribution_hash =
-            phase2::verify_contribution(&params_before, &params_after).expect("invalid transition");
-
-        assert_eq!(
-            &provided_contribution_hash[..],
-            &calculated_contribution_hash[..],
-            "provided contribution hash ({}) does not match calculated contribution hash ({})",
-            hex::encode(&provided_contribution_hash[..]),
-            hex::encode(&calculated_contribution_hash[..]),
-        );
+        })
+    };
 
-        info!(
-            "successfully verified param transition, dt_verify={}s",
-            start_verification.elapsed().as_secs()
+    info!("writing small 'after' params to file: {}", path_after);
```

Comment:
"after" means params after a contribution is made, could you change this log to something like:

```rust
info!("writing raw formatted params to file: {}", path_after);
```

Reply:
Done, but correctly indicating whether raw or non-raw params are being written.